imod/mf6/chd.py

```diff
@@ -141,3 +148,133 @@ def _validate(self, schemata, **kwargs):
+            need to be identical to one of the input grids.
+        regridder_types: RegridMethodType, optional
+            Optional dataclass with regridder types for a specific variable.
+            Use this to override default regridding methods.
+
+        Returns
+        -------
+        A list of Modflow 6 Drainage packages.
+        """
+        return cls._from_head_data(
+            imod5_data[key]["head"],
+            imod5_data["bnd"]["ibound"],
+            target_discretization,
+            regridder_types,
+        )
+
+    @classmethod
+    @standard_log_decorator()
+    def from_imod5_shd_data(
+        cls,
+        imod5_data: dict[str, dict[str, GridDataArray]],
+        target_discretization: StructuredDiscretization,
+        regridder_types: Optional[RegridMethodType] = None,
+    ) -> "ConstantHead":
+        """
+        Construct a ConstantHead-package from iMOD5 data, loaded with the
+        :func:`imod.formats.prj.open_projectfile_data` function.
+
+        This function can be used if no chd packages at all are defined in the imod5 data.
+
+        In that case, imod5 assumed that at all the locations where ibound
+        = -1,  a chd package is active with the starting head of the simulation
+        as a constant.
+
+        So this function creates a single chd package that will be present at all locations where
+        ibound == -1. The assigned head will be the starting head, specified in the array "shd"
+        in the imod5 data.
+
+        Parameters
+        ----------
+        imod5_data: dict
+            Dictionary with iMOD5 data. This can be constructed from the
+            :func:`imod.formats.prj.open_projectfile_data` method.
+        target_discretization:  StructuredDiscretization package
+            The grid that should be used for the new package. Does not
+            need to be identical to one of the input grids.
+        regridder_types: RegridMethodType, optional
+            Optional dataclass with regridder types for a specific variable.
+            Use this to override default regridding methods.
+
+        Returns
+        -------
+        A list of Modflow 6 Drainage packages.
+        """
+        return cls._from_head_data(
+            imod5_data["shd"]["head"],
+            imod5_data["bnd"]["ibound"],
+            target_discretization,
+            regridder_types,
+        )
+
+    @classmethod
+    @standard_log_decorator()
+    def _from_head_data(
+        cls,
+        head: GridDataArray,
+        ibound: GridDataArray,
+        target_discretization: StructuredDiscretization,
+        regridder_types: Optional[RegridMethodType] = None,
+    ) -> "ConstantHead":
+        target_idomain = target_discretization.dataset["idomain"]
+
+        # select locations where ibound < 0
+        head = head.where(ibound < 0)
+
+        # select locations where idomain > 0
+        head = head.where(target_idomain > 0)
```

Comment:
``head`` can be on a different grid than ibound and target_domain. Actually that is a very likely usecase, as the input for the CHD ``head`` usually comes from a different model simulation.

I therefore think it is safer to do this after regridding (also: This reminds me that we do not support not passing the ``regrid_context`` around yet, will make an issue for that).

Reply:
done, now it regrids ibound as well, and it does the "where" statements with the regridded ibound.